### PR TITLE
fix: remove kyc and make text messages to attendee org only

### LINKED ISSUE #11033: [CAL-2477] remove KYC and make SMS outbound enterprise only
after previous hacker incidents using the free SMS workflow we have to put it behind a paywall

sorry.

really annoying a few bad apples screw the rest

<sub>[CAL-2477](https://linear.app/calcom/issue/CAL-2477/remove-kyc-and-make-sms-outbound-enterprise-only)</sub>

### LINKED ISSUE #11443: [CAL-2494] SMS to attendee action is not working for some triggers
The workflow action 'SMS to attendees' is only working for the trigger 'when event is booked' 

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-2494](https://linear.app/calcom/issue/CAL-2494/sms-to-attendee-action-is-not-working-for-some-triggers)</sub>

### PR DESCRIPTION
**Pull request by CarinaWolli**
## What does this PR do?

Makes SMS/Whatsapp to attendees action an org-only feature. The actions SMS/Whatsapp to specific number stay available for all paying users (numbers need to be verified before). With that KYC is not needed anymore and removed. 

Also fixes that 'send SMS to attendee' wasn't working for the triggers 'when rescheduled' and 'when cancelled' due to the missing parameter `eventTypeRequiresConfirmation`.

#### Free user on cal.com:
Screenshot 2023-09-15 at 11 51 02
Screenshot 2023-09-15 at 11 50 11

#### User with team or premium user name: 
Screenshot 2023-09-15 at 11 54 52

#### Org user:
Screenshot 2023-09-15 at 11 52 52

Fixes #11033
Fixes #11443

## Mandatory Tasks

- [x] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

### CODE REVIEW

File: packages/features/ee/workflows/components/WorkflowStepContainer.tsx [hunk 1]
```diff
@@ -331,15 +330,13 @@ export default function WorkflowStepContainer(props: WorkflowStepProps) {
   }
 
   if (step && step.action) {
-    const templateValue = form.watch(`steps.${step.stepNumber - 1}.template`);
```
Comment (CarinaWolli) [inline packages/features/ee/workflows/components/WorkflowStepContainer.tsx:334]: unused code

File: packages/trpc/server/routers/viewer/workflows/sendVerificationCode.handler.ts [hunk 2]
```diff
@@ -10,7 +14,22 @@ type SendVerificationCodeOptions = {
...
+    user && hasKeyInMetadata(user, "isPremium") ? !!user.metadata.isPremium : false;
+
+  let isTeamsPlan = false;
+  if (!isCurrentUsernamePremium) {
+    const { hasTeamPlan } = await hasTeamPlanHandler({ ctx });
+    isTeamsPlan = !!hasTeamPlan;
+  }
+
+  if (!isCurrentUsernamePremium && !isTeamsPlan) {
+    throw new TRPCError({ code: "UNAUTHORIZED" });
```
Comment (CarinaWolli) [inline packages/trpc/server/routers/viewer/workflows/sendVerificationCode.handler.ts:30]: Only paying users should be able to request SMS verification codes

File: packages/trpc/server/routers/viewer/workflows/update.handler.ts [hunk 3]
```diff
@@ -425,11 +424,21 @@ export const updateHandler = async ({ ctx, input }: UpdateOptions) => {
...
+      ) {
+        throw new TRPCError({ code: "UNAUTHORIZED", message: "Not available on free plan" });
       }
-      if (!isKYCVerified && isTextMessageToAttendeeAction(newStep.action)) {
-        throw new TRPCError({ code: "UNAUTHORIZED", message: "Account needs to be verified" });
+      // check if step that require team org already existed before
+      if (
+        !hasOrgsPlan &&
+        isTextMessageToAttendeeAction(newStep.action) &&
+        !isTextMessageToAttendeeAction(oldStep.action)
```
Comment (CarinaWolli) [inline packages/trpc/server/routers/viewer/workflows/update.handler.ts:439]: Only throw an unauthorized error if the edited step was not a text message to attendees already before. All existing workflow should continue to work

File: packages/trpc/server/routers/viewer/workflows/update.handler.ts [hunk 4]
```diff
@@ -605,11 +614,11 @@ export const updateHandler = async ({ ctx, input }: UpdateOptions) => {
...
   const addedSteps = steps.map((s) => {
     if (s.id <= 0) {
-      if (isSMSOrWhatsappAction(s.action) && !hasPaidPlan) {
-        throw new TRPCError({ code: "UNAUTHORIZED" });
+      if (isSMSOrWhatsappAction(s.action) && !isTextMessageToAttendeeAction(s.action) && !hasPaidPlan) {
+        throw new TRPCError({ code: "UNAUTHORIZED", message: "Not available on free plan" });
       }
-      if (!isKYCVerified && isTextMessageToAttendeeAction(s.action)) {
-        throw new TRPCError({ code: "UNAUTHORIZED", message: "Account needs to be verified" });
+      if (!hasOrgsPlan && isTextMessageToAttendeeAction(s.action)) {
```
Comment (CarinaWolli) [inline packages/trpc/server/routers/viewer/workflows/update.handler.ts:620]: throw unauthorized if a non org users tries to add a 'text message to attendee' action

File: packages/ui/components/badge/UpgradeOrgsBadge.tsx [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+
+import { Tooltip } from "../tooltip";
+import { Badge } from "./Badge";
+
+export const UpgradeOrgsBadge = function UpgradeOrgsBadge() {
+  const { t } = useLocale();
+
+  return (
+    <Tooltip content={t("orgs_upgrade_to_enable_feature")}>
+      <a href="https://cal.com/enterprise" target="_blank">
```
Comment (CarinaWolli) [inline packages/ui/components/badge/UpgradeOrgsBadge.tsx:11]: are we ok with linking to https://cal.com/enterprise when clicking on the upgrade badge? 
Screenshot 2023-09-15 at 11 43 23

File: packages/trpc/server/routers/viewer/workflows/update.handler.ts (hunk 4 above)
Comment (CarinaWolli) [inline packages/trpc/server/routers/viewer/workflows/update.handler.ts:620]: throw unauthorized if non org user tries to add a 'text message to attendee' action

File: packages/trpc/server/routers/viewer/workflows/update.handler.ts (hunk 3 above)
Comment (CarinaWolli) [inline packages/trpc/server/routers/viewer/workflows/update.handler.ts:439]: Only throw an unauthorized error if the edited step was not a 'text message to attendee' action already before

File: packages/features/bookings/lib/handleCancelBooking.ts [hunk 6]
```diff
@@ -67,23 +66,7 @@ async function getBookingToDelete(id: number | undefined, uid: string | undefine
...
-                  team: {
-                    select: {
-                      metadata: true,
-                    },
-                  },
-                },
-              },
-            },
-          },
+          owner: true,
```
Comment (alannnc) [inline packages/features/bookings/lib/handleCancelBooking.ts:69]:
```suggestion
          owner: {
              select: {
                  id: true,
                  hideBranding: true,
                  metadata: true,
              },
```

**Review by alannnc — DISMISSED** [review]
Code looks good and I tested the scenarios described. 👍

**Review by alannnc — APPROVED** [review]

**Pull request merged**